File: app/src/main/java/com/example/daypilot_test_desing/core/data/repository/TaskRepository.kt
package com.example.daypilot_test_desing.core.data.repository

import com.example.daypilot_test_desing.core.data.model.CalendarTaskData
import com.example.daypilot_test_desing.core.data.model.NewTaskData
import com.example.daypilot_test_desing.core.data.model.TaskCategory
import com.example.daypilot_test_desing.core.data.model.TaskDifficulty

interface TaskRepository {
    suspend fun getTasks(): List<CalendarTaskData>
    suspend fun addTask(data: NewTaskData)
    suspend fun updateTask(id: String, title: String, category: TaskCategory, difficulty: TaskDifficulty, duration: Int, description: String = "")
    suspend fun toggleTask(occurrenceId: String, isDone: Boolean)
    suspend fun deleteTask(id: String)
}

File: lib/core/data/repositories/task_repository.dart
import '../models/app_task.dart';
import '../models/task_category.dart';
import '../models/task_difficulty.dart';

abstract class TaskRepository {
  Future<List<AppTask>> getTasks();

  Future<void> addTask(NewTaskData data);

  Future<void> updateTask({
    required String id,
    required String title,
    required String description,
    required TaskCategory category,
    required TaskDifficulty difficulty,
    required int durationMinutes,
  });

  Future<void> toggleTask({required String occurrenceId, required bool isDone});

  Future<void> deleteTask(String id);
}
